### PR TITLE
Make users confirm the deletion of the initial task node

### LINKED ISSUE #2090: Confirm deletion of initial task node
See https://discuss.webknossos.org/t/protect-the-initial-node-from-being-deleted/539

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/2090/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by daniel-wer**
From https://discuss.webknossos.org/t/protect-the-initial-node-from-being-deleted/539/

> I think we should do this:
>  show a message that needs to be confirmed if the user tries to delete the node with the task ID1 (I guess this is always the ID if we create tasks with given coordinates?)
> the tracing needs to be part of a task (no explorative please because exploratives are only done by scientists or very experienced hiwis).
> Suggested message: "Do you really want to delete the initial node?"
> 
> if the user tries to delete a tree that contains the initial node there should ideally be a different tree-deletion warning: This tree contains the initial node. Do you really want to delete the whole tree?

### Mailable description of changes:
 - The deletion of the initial node of a task (or of the tree that contains that initial node) now needs to be confirmed explicitly by the user.

### Steps to test:
- Open a task and try to delete the initial node or tree. Confirms should be shown in each case.
- Delete other nodes, no confirm should be shown.
- Delete a tree that is not the initial tree, a different confirm should be shown.
- Nothing should have changed in explorative tracings.

### Issues:
- fixes #2090

------
- [x] Ready for review

### CODE REVIEW

File: app/assets/javascripts/oxalis/model/actions/skeletontracing_actions.js [hunk 1]
```diff
@@ -152,6 +161,25 @@ export const deleteNodeAction = (
...
+  nodeId?: number,
+  treeId?: number,
+): DeleteNodeActionType | NoActionType => {
+  const state = Store.getState();
+  return getActiveNode(state.tracing)
+    .map(activeNode => {
+      nodeId = nodeId != null ? nodeId : activeNode.id;
+      if (state.task != null && nodeId === 1) {
+        // Let the user confirm the deletion of the initial node (node with id 1) of a task
+        if (!Window.confirm("Do you really want to delete the initial node?")) {
```
Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/model/actions/skeletontracing_actions.js]: I wonder if we could use a nice and shiny Antd `confirm` here? ✨ https://ant.design/components/modal/#components-modal-demo-confirm
Also, we we should store all messages in the `messages.js`.

File: app/assets/javascripts/oxalis/model/actions/skeletontracing_actions.js [hunk 2]
```diff
@@ -212,6 +240,30 @@ export const deleteTreeAction = (
...
+export const deleteTreeWithConfirmAction = (
+  treeId?: number,
+): DeleteTreeActionType | NoActionType => {
+  const state = Store.getState();
+  return getTree(state.tracing, treeId)
+    .map(tree => {
+      if (state.task != null && tree.nodes[1] != null) {
+        // Let the user confirm the deletion of the initial node (node with id 1) of a task
+        if (
+          !Window.confirm(
```
Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/model/actions/skeletontracing_actions.js]: The same here

File: app/assets/javascripts/oxalis/model/actions/skeletontracing_actions.js [hunk 3]
```diff
@@ -212,6 +240,30 @@ export const deleteTreeAction = (
...
+            "This tree contains the initial node. Do you really want to delete the whole tree?",
+          )
+        ) {
+          return noAction();
+        }
+      } else if (!Window.confirm("Do you really want to delete the whole tree?")) {
```
Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/model/actions/skeletontracing_actions.js]: And here.

**Comment by daniel-wer** [comment]
@hotzenklotz As the task creation works again, do you still want to do a manual test?

**Comment by hotzenklotz** [comment]
Works like a charm. 
giphy-downsized

**Pull request merged**